graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -99,7 +99,7 @@ def _validate_non_required_macro_definition_directives(
         inside_fold_scope: bool, whether the subtree is within a @fold scope
     """
     names_of_allowed_directives = frozenset(
-        {directive.name for directive in DIRECTIVES_ALLOWED_IN_MACRO_EDGE_DEFINITION}
+        {directive for directive in DIRECTIVES_ALLOWED_IN_MACRO_EDGE_DEFINITION}
```

Comment:
I think `DIRECTIVES_ALLOWED_IN_MACRO_EDGE_DEFINITION` is already a frozenset of the names of those directives, so I think `names_of_allowed_directives` is unnecessary as a whole — the global constant should be fine to use throughout.